src/Core/Rexl.Bind/Operations/Funcs/Text.cs

```diff
@@ -758,3 +758,123 @@ public static string Exec(string src, string remove, string insert)
+/// </summary>
+public sealed partial class TextPadFunc : RexlOper
+{
+    public enum PadKind : byte
+    {
+        // Trim both leading and trailing whitespace.
+        Center,
+        // Trim only leading whitespace.
+        Start,
+        // Trim only trailing whitespace.
+        End
+    }
+
+    public static readonly TextPadFunc PadCenter = new TextPadFunc(PadKind.Center, "PadCenter");
+    public static readonly TextPadFunc PadStart = new TextPadFunc(PadKind.Start, "PadStart");
+    public static readonly TextPadFunc PadEnd = new TextPadFunc(PadKind.End, "PadEnd");
+
+    public PadKind Kind { get; }
+
+    public Func<string, long, string> Map { get; }
+
+    private TextPadFunc(PadKind kind, string name)
+        : base(isFunc: true, new DName(name), BindUtil.TextNs, 2, 2)
+    {
+        switch(kind) {
+        case PadKind.Start:
+            Map = ExecStart;
+            break;
+        case PadKind.End:
+            Map = ExecEnd;
+            break;
+        case PadKind.Center:
+        default:
+            Map = ExecCenter;
+            break;
+        }
+
+        Kind = kind;
+    }
+
+    protected override ArgTraits GetArgTraitsCore(int carg)
+    {
+        Validation.Assert(SupportsArity(carg));
+        var maskAll = BitSet.GetMask(carg);
+        var maskOpt = maskAll.ClearBit(0);
+        return ArgTraitsLifting.Create(this, carg, maskLiftSeq: maskAll, maskLiftTen: maskAll, maskLiftOpt: maskOpt);
+    }
+
+    protected override (DType, Immutable.Array<DType>) SpecializeTypesCore(InvocationInfo info)
+    {
+        Validation.AssertValue(info);
+        Validation.Assert(SupportsArity(info.Arity));
+        Validation.Assert(info.Arity == 2);
+
+        return (DType.Text, Immutable.Array.Create(DType.Text, DType.I8Req));
+    }
+
+    protected override bool CertifyCore(BndCallNode call, ref bool full)
+    {
+        if (call.Type != DType.Text)
+            return false;
+        var args = call.Args;
+        if (args[0].Type != DType.Text)
+            return false;
+        if (args[1].Type != DType.I8Req)
+            return false;
+        return true;
+    }
+
+    protected override BoundNode ReduceCore(IReducer reducer, BndCallNode call)
+    {
+        Validation.AssertValue(reducer);
+        Validation.Assert(IsValidCall(call));
+
+        var srcArg = call.Args[0];
+        var lenArg = call.Args[1];
+        if (srcArg.TryGetString(out var str) && lenArg.TryGetI8(out var len ))
+        {
+            if (string.IsNullOrEmpty(str) || len == 0)
+                return srcArg;
+            return BndStrNode.Create(Map(str, len));
+        }
+
+        return call;
+    }
+
+    public static string ExecCenter(string src, long len) {
+        if (len <= 0)
+            return src;
+        int count = (int)Math.Min(len, int.MaxValue);
+        if (string.IsNullOrEmpty(src))
+            return new string(' ', count);
+        return src.PadLeft(count).PadLeft(count);
```

Comment:
This isn't correct. We want the result to be the given length. Half the characters should go one one side and the other half on the other. I don't think we'll want to use the existing .Net pad functions to do this.